components/user/UserProfile.js

```diff
@@ -68,34 +69,77 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       </div>
       <div className="flex justify-center my-4 text-center">
-        <ReactMarkdown components={{ a: LinkRenderer }}>{data.bio}</ReactMarkdown>
+        <ReactMarkdown components={{ a: LinkRenderer }}>
+          {data.bio}
+        </ReactMarkdown>
       </div>
       {!qrShow && (
         <div className="flex flex-wrap justify-center">
           {data.tags &&
             data.tags.map((tag) => (
-              <Tag name={tag} key={tag.toLowerCase()} onClick={() => router.push(`/search?keyword=${tag.toLowerCase()}`)} />
+              <Tag
+                name={tag}
+                key={tag.toLowerCase()}
+                onClick={() =>
+                  router.push(`/search?keyword=${tag.toLowerCase()}`)
+                }
+              />
             ))}
         </div>
       )}
 
       {/* Passed Ref object as the ref attribute to the JSX of the DOM node of QR */}
-      <div className="flex justify-center my-4" ref={qrRef}>
-        {qrShow && (
-          <QRCodeCanvas
-            className="border border-white"
-            value={`${BASE_URL}/${data.username}`}
-            size={fallbackImageSize * 2}
-          />
-        )}
-      </div>
-      <div className="flex justify-center mb-4">
-        {qrShow && (
-          <Button primary={true} onClick={downloadQR}>
-            Download QR code
-          </Button>
-        )}
-      </div>
+      <Modal show={qrShow} setShow={setQrShow} modalStyles="w-fit m-auto">
+        <div className="flex flex-col items-center justify-center px-8">
+          <div>
+            <div className="flex justify-center my-4" ref={qrRef}>
+              {qrShow && (
+                <QRCodeCanvas
+                  className="border border-white"
+                  value={`${BASE_URL}/${data.username}`}
+                  size={fallbackImageSize * 2}
+                />
+              )}
+            </div>
+            <div className="w-[100%] px-2 mx-auto flex justify-center mb-4">
+              {qrShow && (
+                <Button primary={true} onClick={downloadQR}>
+                  Download QR code
+                </Button>
+              )}
+            </div>
+          </div>
+          {qrShow && (
+            <>
+              <div className="h-full m-4 p-2 flex flex-row items-start justify-center space-x-2">
+                {socials.map(({ href, Icon, includeText }) => (
+                  <a
```

Comment:
We should be using our `Link` component here

Reply:
For external redirects also?

Reply:
Yep 👍  you can pass `target="_blank"` as a prop

Reply:
It will pass the additional props here https://github.com/EddieHubCommunity/LinkFree/blob/main/components/Link.js#L13

We use it in the footer also https://github.com/EddieHubCommunity/LinkFree/blob/fbd91166d0abc3b4d65bd698e1d7275e4c9f8a67/components/Footer.js#L116